### ISSUE TITLE
pylint trigger false-positive `no-member` error when using `pathlib.Path` parents function 

### ISSUE THREAD

**Issue by trym-b** [PyCQA/pylint#5783]
### Bug description

Pylint finds a false-positive when running the code below:
```python
from pathlib import Path


def _main() -> None:
    current_path = Path().resolve()
    parent_path = current_path.parents[2].resolve()
    print(f"{current_path=}")
    print(f"{parent_path=}")


if __name__ == "__main__":
    _main()
```
When the code is run like this: `python /path/to/script/test.py`, this is the output:
```
$ python test.py
current_path=PosixPath('/home/user_name/testing/pylint-no-member')
parent_path=PosixPath('/home')
```

However, when I run pylint with the following options `pylint test.py --disable missing-module-docstring` then this is the output:
```
pylint test.py --disable missing-module-docstring
************* Module test
test.py:6:18: E1101: Instance of 'tuple' has no 'resolve' member (no-member)

------------------------------------------------------------------
Your code has been rated at 3.75/10 (previous run: 3.75/10, +0.00)
```

### Command used

```shell
pylint test.py --disable missing-module-docstring
```

### Pylint output

```shell
pylint test.py --disable missing-module-docstring
************* Module test
test.py:6:18: E1101: Instance of 'tuple' has no 'resolve' member (no-member)

------------------------------------------------------------------
Your code has been rated at 3.75/10 (previous run: 3.75/10, +0.00)
```

### Expected behavior

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.2 (main, Jan 15 2022, 18:02:07) [GCC 9.3.0]
```

### OS / Environment

Ubuntu 20.04

**Comment by stephanlachnit** [comment]
Interestingly this only happens with Python 3.10, not with Python 3.9.

**Comment by deepyaman** [comment]
Same as #5832. You can resolve this particular error by now by doing:

```python
parent_path = current_path.parents[2].resolve()
```

**Comment by galah92** [comment]
I'm getting the same issue with the following code:
```python
cwd = Path(__file__).resolve().parents[1]
apps_paths = cwd.glob('apps/*')
```
On Python 10 as well. Is there an expected fix?
